### PR TITLE
#309 -  Message navigation

### PR DESCRIPTION
**Pull request by jojopirker**
Draft for an implementation for #309. 

Looks like this currently:
https://user-images.githubusercontent.com/19796430/210849009-df5c03d8-d67d-4646-8091-64e555d49aff.mov

### CODE REVIEW

File: website/src/pages/api/messages/[id]/parent.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  const { id } = req.query;
+
+  if (!id) {
+    res.status(400).end();
+    return;
+  }
+
+  const messageRes = await fetch(`${process.env.FASTAPI_URL}/api/v1/messages/${id}`, {
```
Comment (fozziethebeat) [inline website/src/pages/api/messages/[id]/parent.ts:19]: Should this include `/parent` in the URL?

**Review by fozziethebeat — COMMENTED** [review]
Broadly this looks right! Tidy up as much as you think needed and I'll do a more careful review.

File: website/src/pages/api/messages/[id]/parent.ts (hunk 1 above)
Comment (jojopirker) [reply website/src/pages/api/messages/[id]/parent.ts:19]: I believe the backend does not provide the parent function..

Comment (fozziethebeat) [reply website/src/pages/api/messages/[id]/parent.ts:19]: I read this a bit too quickly before going to take a bath.  You are right.  This seems to be the only way right now

**Comment by jojopirker** [comment]
made a few changes to make it recursive... 
Looks the same if message tree is not big, but still has some visual quirks with a bigger trees (see attached video)

@fozziethebeat maybe you got some ideas how to properly style it...
I would leave it as a draft still but I think you could already take a look at the code, please :)

https://user-images.githubusercontent.com/19796430/211035618-0bb5383a-9bfd-44dd-a556-1627a6b15569.mov

**Comment by fozziethebeat** [comment]
I don't see many code suggestions to make.  It's clean and pretty easy to understand.  Later I think we can fold the backend fetch calls into a library someone is working on but I'm also trying to negotiate not having these API routes anyways so it's not a huge issue.

Will approve once you feel it's ready

**Comment by jojopirker** [comment]
should be ready now :)

Screenshot 2023-01-07 at 10 23 59

**Review by fozziethebeat — APPROVED** [review]

**Comment by AbdBarho** [comment]
This actually looks sick!!!

**Pull request merged**